### PR TITLE
Add code-generator for track2 client and sample clients

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by lzhecheng** [comment]
/retest
last failure seems testing framework bug

**Review by lzhecheng — COMMENTED** [review]
Some nits.

File: pkg/azureclients/v2/client_config.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// ClientConfig contains all essential information to create an Azure client.
+type ClientConfig struct {
+	CloudName               string
+	Location                string
+	ExtendedLocation        *ExtendedLocation
+	SubscriptionID          string
+	ResourceManagerEndpoint string
+	RateLimitConfig         *policy.RateLimitConfig
+	RestClientConfig        *policy.RetryConfig
+	//	UserAgent               string
```
[the remark this answers is not in the document]
Comment (MartinForReal) [reply pkg/azureclients/v2/client_config.go]: removed。Thanks！

**Pull request merged**